pages/index.client.vue

```diff
@@ -0,0 +1,156 @@
+<script setup lang="ts">
+import {
+  AisInstantSearch,
+  AisSearchBox,
+  AisHits,
+  AisRefinementList,
+} from "vue-instantsearch/vue3/es";
+
+const indexName = "test_permit_fees";
+const algolia = useAlgoliaRef();
+
+const transformCategories =
+  (taxonomy_uid: string) =>
+  (
+    items: ToDo[],
+    { results }: { results: { hits: AlgoliaProjectEntry[] } },
+  ) => {
+    const categories = new Map<string, boolean>(
+      results.hits
+        .flatMap((x) => x.taxonomies)
+        .filter((x) => x.taxonomy_uid === taxonomy_uid)
+        .map((x) => [x.term_uid, true]),
+    );
+
+    function getTermNameFromKey(key: string) {
```

Comment:
@JordanPisani  let's move the `getTermNameFromKey` function up out of the `transformCategories` scope (line 11?). Every time `transformCategories` is called you're creating a new instance of the `getTermNameFromKey` function